Corona-Warn-App/src/test/java/de/rki/coronawarnapp/nearby/windows/ExposureWindowsCalculationTest.kt

```diff
@@ -0,0 +1,405 @@
+        val result = StringBuilder()
+        result.append("\n\n").append("------------ $title -----------")
+        result.append("\n").append("Mocked Exposure window: #${exposureWindow.hashCode()}")
+        result.append("\n").append("◦ Calibration Confidence: ${exposureWindow.calibrationConfidence}")
+        result.append("\n").append("◦ Date Millis Since Epoch: ${exposureWindow.dateMillisSinceEpoch}")
+        result.append("\n").append("◦ Infectiousness: ${exposureWindow.infectiousness}")
+        result.append("\n").append("◦ Report type: ${exposureWindow.reportType}")
+
+        result.append("\n").append("‣ Scan Instances (${exposureWindow.scanInstances.size}):")
+        for (scan: ScanInstance in exposureWindow.scanInstances) {
+            result.append("\n\t").append("⇥ Mocked Scan Instance: #${scan.hashCode()}")
+            result.append("\n\t\t").append("↳ Min Attenuation: ${scan.minAttenuationDb}")
+            result.append("\n\t\t").append("↳ Seconds Since Last Scan: ${scan.secondsSinceLastScan}")
+            result.append("\n\t\t").append("↳ Typical Attenuation: ${scan.typicalAttenuationDb}")
+        }
+        result.append("\n").append("-------------------------------------------- ✂ ----").append("\n")
+        debugLog(result.toString(), logLevel)
+    }
+
+    private fun jsonToConfiguration(json: DefaultRiskCalculationConfiguration) {
+
+        testConfig = ConfigDataContainer(
+            serverTime = Instant.now(),
+            localOffset = Duration.ZERO,
+            mappedConfig = configData,
+            identifier = "soup",
+            configType = ConfigData.Type.FROM_SERVER
+        )
+
+        val attenuationFilters = mutableListOf<RiskCalculationParametersOuterClass.MinutesAtAttenuationFilter>()
+        for (jsonFilter: JsonMinutesAtAttenuationFilter in json.minutesAtAttenuationFilters) {
+            val filter: RiskCalculationParametersOuterClass.MinutesAtAttenuationFilter = mockk()
+            every { filter.attenuationRange.min } returns jsonFilter.attenuationRange.min
+            every { filter.attenuationRange.max } returns jsonFilter.attenuationRange.max
+            every { filter.attenuationRange.minExclusive } returns jsonFilter.attenuationRange.minExclusive
+            every { filter.attenuationRange.maxExclusive } returns jsonFilter.attenuationRange.maxExclusive
+            every { filter.dropIfMinutesInRange.min } returns jsonFilter.dropIfMinutesInRange.min
+            every { filter.dropIfMinutesInRange.max } returns jsonFilter.dropIfMinutesInRange.max
+            every { filter.dropIfMinutesInRange.minExclusive } returns jsonFilter.dropIfMinutesInRange.minExclusive
+            every { filter.dropIfMinutesInRange.maxExclusive } returns jsonFilter.dropIfMinutesInRange.maxExclusive
+            attenuationFilters.add(filter)
+        }
+        every { testConfig.minutesAtAttenuationFilters } returns attenuationFilters
```

Comment:
Check this please, does this actually work? Mocking on a non mock?

Reply:
Nvm. "mappedConfig = configData" is a mock and the interface is forwarding here. Works as expected.